sdks/python/apache_beam/io/fileio.py

```diff
@@ -835,10 +839,15 @@ def finish_bundle(self):
 
 
 class _RemoveDuplicates(beam.DoFn):
-
+  """Internal DoFn that filters out filenames already seen (even though the file
+  has updated)."""
   COUNT_STATE = CombiningValueStateSpec('count', combine_fn=sum)
 
-  def process(self, element, count_state=beam.DoFn.StateParam(COUNT_STATE)):
+  def process(
+      self,
+      element: Tuple[str, filesystem.FileMetadata],
```

Comment:
Got warnings 
```
WARNING:apache_beam.transforms.core:Key coder FastPrimitivesCoder for transform <ParDo(PTransform) label=[RemoveAlreadyRead]> with stateful DoFn may not be deterministic.
This may cause incorrect behavior for complex key types.
Consider adding an input type hint for this transform.

WARNING:apache_beam.transforms.core:Key coder FastPrimitivesCoder for transform <ParDo(PTransform) label=[RemoveOldAlreadyRead]> with stateful DoFn may not be deterministic.
This may cause incorrect behavior for complex key types.
Consider adding an input type hint for this transform.
```
tried these type hints but the warning persist. This may be relevant to flaky tests. How can I add typehint properly and resolve these warnings?